src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskDeletionService.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.services;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.query.TaskResourceCaseQueryBuilder;
+
+import java.util.List;
+
+import static uk.gov.hmcts.reform.wataskmanagementapi.cft.enums.CFTTaskState.TERMINATED;
+import static uk.gov.hmcts.reform.wataskmanagementapi.services.utils.DeleteTasksServiceHelper.getTaskIds;
+
+@Slf4j
+@Service
+public class TaskDeletionService {
+
+    private final CFTTaskDatabaseService cftTaskDatabaseService;
+
+    @Autowired
+    public TaskDeletionService(final CFTTaskDatabaseService cftTaskDatabaseService) {
+        this.cftTaskDatabaseService = cftTaskDatabaseService;
+    }
+
+    public void deleteTasksByCaseId(final String caseId) {
+        final List<TaskResourceCaseQueryBuilder> taskResourceCaseQueryBuilders = cftTaskDatabaseService
+                .findByTaskIdsByCaseId(caseId);
+
+        filterAllUnterminatedTasksAndLogError(taskResourceCaseQueryBuilders, caseId);
+
+        deleteTasks(taskResourceCaseQueryBuilders, caseId);
+    }
+
+    private void deleteTasks(final List<TaskResourceCaseQueryBuilder> taskResourceCaseQueryBuilders,
+                             final String caseId) {
+        try {
+            cftTaskDatabaseService.deleteTasks(getTaskIds(taskResourceCaseQueryBuilders));
+        } catch (final Exception exception) {
+            log.error(String.format("Unable to delete all tasks for case id: %s", caseId));
```

Comment:
Please see missing coverage for new lines as discussed - > https://sonarcloud.io/component_measures?metric=new_coverage&selected=uk.gov.hmcts.reform%3Awa-task-management-api%3Asrc%2Fmain%2Fjava%2Fuk%2Fgov%2Fhmcts%2Freform%2Fwataskmanagementapi%2Fservices%2FTaskDeletionService.java&view=list&pullRequest=991&id=uk.gov.hmcts.reform%3Awa-task-management-api

Reply:
Done